tests/storages_tests/test_in_memory.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+
+import optuna
+from optuna.storages import InMemoryStorage
+
+
+@pytest.mark.parametrize('study_direction', ['minimize', 'maximize'])
+def test_update_cache(study_direction):
+    # type: (str) -> None
+
+    storage = InMemoryStorage()
+
+    assert storage.best_trial_id is None
+
+    # If the direction is 'minimize', the objective function is weakly decreasing.
+    # Otherwise, it is weakly increasing.
+    sign = -1 if study_direction == 'minimize' else 1
+
+    study = optuna.create_study(storage=storage, direction=study_direction)
+    study.optimize(lambda x: sign * x.number, n_trials=1)
```

Comment:
LGTM! On a picky note, I'd personally name the lambda argument `t` or `trail` for readability and consistency with the other tests since `x` is quite general, but again a minor.

Reply:
You're right. I'll fix it in this PR. Thank you!

Reply:
Fixed: https://github.com/pfnet/optuna/pull/611/commits/86dfaaa4f1ff77ac4669ed1962fa76396cb95524.

Reply:
Thanks for the quick response, I'll merge it once CI finishes!